### PR TITLE
#1912 trap, exception on adding duplicating hero

### PR DESCRIPTION
**Pull request by nullkiller**
A trap for #1912. We expect it is duplicating heroes issue so I added a trap for it to trigger earlier and reveal who wants to add duplicating hero. The map with duplicating Orrins attached to #1912 crashes on start now as expected

### CODE REVIEW

**Comment by Nordsoft91** [comment]
Make sense to add check for same heroes into map editor validation so mapmaker will be able to understand what's wrong with the map (if it's not there)

File: lib/mapObjects/CGHeroInstance.cpp [hunk 1]
```diff
@@ -1489,8 +1489,22 @@ std::string CGHeroInstance::getHeroTypeName() const
...
+	auto existingHero = std::find_if(map->objects.begin(), map->objects.end(), [&](const CGObjectInstance * o) ->bool
+		{
+			return (o->ID == Obj::HERO || o->ID == Obj::PRISON) && o->subID == subID && o != this;
+		});
+
+	if(existingHero != map->objects.end())
+	{
+		logGlobal->error("Hero is already on the map at %s", (*existingHero)->visitablePos().toString());
+
+		throw std::exception("Hero is already on the map");
```
Comment (IvanSavenko) [inline lib/mapObjects/CGHeroInstance.cpp:1501]: To fix CI build
```suggestion
		throw std::runtime_error("Hero is already on the map");
```

**Comment by nullkiller** [comment]
Actually will check if this thing can crash map editor

**Comment by nullkiller** [comment]
It crashes map editor. So need to find a way to disable this check there as it has own validation for this.

**Comment by nullkiller** [comment]
Added session variable to tell if we are inside map editor. Now in editor it just issues warning and editor also has validation rule for duplicating heroes

**Comment by IvanSavenko** [comment]
Then I guess let's merge and see if it will detect issues like #3099 ?

**Pull request merged**